### PR TITLE
Merge missing compositionalIT bits

### PR DESCRIPTION
**Pull request by nargiz**
This PR closes #

The changes in this PR are as follows:

* we were missing some changes from CompositinalIT for Zone Redundancy. Merged with latest CompositionalIT

I have read the [contributing guidelines](CONTRIBUTING.md) and have completed the following:

* [ ] **Tested my code** end-to-end against a live Azure subscription.
* [ ] **Updated the documentation** in the docs folder for the affected changes.
* [ ] **Written unit tests** against the modified code that I have made.
* [ ] **Updated the [release notes](RELEASE_NOTES.md)** with a new entry for this PR.
* [ ] **Checked the coding standards** outlined in the [contributions guide](CONTRIBUTING.md) and ensured my code adheres to them.

If I haven't completed any of the tasks above, I include the reasons why here:

Below is a minimal example configuration that includes the new features, which can be used to deploy to Azure:

```fsharp
```

### CODE REVIEW

File: src/Farmer/Builders/Builders.WebApp.fs [hunk 1]
```diff
@@ -736,6 +736,13 @@ type WebAppBuilder() =
     member this.AddCustomDomains(state, customDomains:string list) = customDomains |> List.fold (fun state domain -> this.AddCustomDomain(state, domain)) state
     member this.AddCustomDomains(state, domainConfigs:DomainConfig list) = domainConfigs |> List.fold (fun state domain -> this.AddCustomDomain(state, domain)) state
     member this.AddCustomDomains(state, customDomainsWithThumprint:(string * ArmExpression) list) = customDomainsWithThumprint |> List.fold (fun state domain -> this.AddCustomDomain(state, domain)) state
+    
+    /// Map specified port traffic from your docker container to port 80 for App Service
+    [<CustomOperation "docker_port">]
+    member _.DockerPort(state: WebAppConfig, dockerPort:int) = { state with DockerPort = Some dockerPort }
+    /// Enables the zone redundancy in service plan
+    [<CustomOperation "zone_redundant">]
+    member this.ZoneRedundant(state:WebAppConfig, flag:FeatureFlag) = {state with ZoneRedundant = Some flag}
```
Comment (BlakeWills) [inline src/Farmer/Builders/Builders.WebApp.fs:745]: Does this only work for webapps? What about functions?

**Comment by TheRSP** [comment]
@JustinWilkinson didn't you merge recently? Does this duplicate/conflict with your merge (it seems like a lot of commits for 2 days of changes)

File: src/Farmer/Builders/Builders.WebApp.fs (hunk 1 above)
Comment (nargiz) [reply src/Farmer/Builders/Builders.WebApp.fs:745]: ZoneRedundancy is an app service plan setting. When farmer creates a function if asp is not specified it creates the one with Dynamic SKU (for some reason hardcoded) which doesn't support zone redundancy. I have noticed that in web apps we can specify SKU, so added also ZoneRedundancy support. But we shouldn't be worried about it as we never create asp as part of the web app or function creation. In our circumstances, we will specify ZoneRedundant when we create asp. Then link web app/functions to use it. That part was already merged.

**Comment by nargiz** [comment]
> @JustinWilkinson didn't you merge recently? Does this duplicate/conflict with your merge (it seems like a lot of commits for 2 days of changes)

It could be missed when we were resolving merge conflicts. Sorry, my slot settings PR is stuck in the original farmer and creates a pain in the merge :( I will sort it one day, hopefully soon :(

**Comment by nargiz** [comment]
changes are already merged in another PR

**Pull request closed without merging**